### PR TITLE
v2v: add a new case for converting a guest to a local block device

### PR DESCRIPTION
**Pull request by xiaodwan**
See: polarion_id#RHEL-241703 and bz#1868690

Signed-off-by: Xiaodai Wang <xiaodwan@redhat.com>

### CODE REVIEW

**Comment by xiaodwan** [comment]
(1/1) type_specific.io-github-autotest-libvirt.function_test_esx.negative_test.block_dev.small.esx_70.json.pattern.it_vddk.vpx_uri: PASS (121.17 s)
(1/1) type_specific.io-github-autotest-libvirt.function_test_esx.positive_test.block_dev.large.esx_70.json.pattern.it_vddk.vpx_uri: PASS (409.11 s)

**Comment by xiaodwan** [comment]
depends on https://github.com/avocado-framework/avocado-vt/pull/3088

**Comment by xiaodwan** [comment]
@chunfuwen @chloerh @yantian0 Please help to review this PR, thanks.

File: v2v/tests/src/function_test_esx.py [hunk 1]
```diff
@@ -823,6 +827,31 @@ def vm_check(status_error):
...
+            os_directory = tempfile.TemporaryDirectory(prefix='v2v_test_', dir=os_directory)
+            diskimage = '%s/diskimage' % os_directory.name
+            # Update 'os_directory' for '-os' option
+            params['os_directory'] = os_directory.name
+
+            # Create a 1G image
+            cmd = 'dd if=/dev/zero of=%s bs=10M count=%s' % (diskimage, block_count)
+            process.run(cmd, shell=True)
+            # Build filesystem
+            cmd = 'mkfs.ext4 %s' % diskimage
```
Comment (yantian0) [inline v2v/tests/src/function_test_esx.py:842]: XFS is the default file system in RHEL. Why not use mkfs.xfs insteading mkfs.ext4?

File: v2v/tests/cfg/function_test_esx.cfg [hunk 2]
```diff
@@ -475,6 +476,19 @@
...
+            main_vm = VM_NAME_ESX70_RHEL8_V2V_EXAMPLE
+            checkpoint = "block_dev"
+            os_directory = '/var/lib/libvirt/images/'
+            json_disk_pattern = 'mydisk%{DiskNo}'
+            variants:
+                - small:
+                    only negative_test
+                    block_count = 103
+                - large:
+                    block_count = 1300
```
Comment (yantian0) [inline v2v/tests/cfg/function_test_esx.cfg:491]: Where will invoke block_dev.large? Just be curious why it is 1300.

File: v2v/tests/src/function_test_esx.py (hunk 1 above)
Comment (xiaodwan) [reply v2v/tests/src/function_test_esx.py:842]: Not all rhel version. This is not a key step, it doesn't matter.

File: v2v/tests/cfg/function_test_esx.cfg (hunk 2 above)
Comment (xiaodwan) [reply v2v/tests/cfg/function_test_esx.cfg:491]: This is a postitive case. the block_count = 1300 will be passed to 'dd' and allocate a ~13G disk image.

**Review by yantian0 — APPROVED** [review]

**Comment by xiaodwan** [comment]
@chloerh Please help to review this PR, thanks

**Pull request merged**